### PR TITLE
sci-misc/boinc: Updates to deal with bugs 584386, 599910 and 603522

### PR DESCRIPTION
**Pull request by Yamakuzure**
This pull request updates the following issues and (hopefully) closes Gentoo-Bugs 584386, 599910 and 603522.

-  start() no longer uses the --daemon option of the boinc_client, but the --background option of the start-stop-daemon command. Further it creates a pid file in the path set by the new config variable BOINC_PIDFILE, that has been added to boinc.conf.
- stop() no longer uses boinccmd to send a quit signal, but uses the --stop and --pidfile options of the start-stop-daemon command. The waiting time should be large enough to successfully await the end of the exiting task of the boinc_client program.
- cuda_check() now checks the validity of the libcudart.so symlink and removes it if it is invalid. Further it looks for a present libcudart.so library in /opt/cuda/lib* and picks the newest found to create a new symlink if none is present.
- The suspend() and resume() functions have been updated to use the start-stop-daemon command, so both the user:group and a possibly required password are now used to circumvent authentication errors.
- The old and obsolete ebuilds have been removed.
- Revision bump of the current ebuilds to enforce the usage of the new init script and config.

Further a hotfix has been added to deal with Gentoo-Bug 599910:
- The desktop file is renamed from boincmgr-boinc.desktop to boincmgr.desktop. The GNOME shell seems to need it that way.

The security issue pointed out in Gentoo-Bug 603522 has been dealt with:
- The init script no longer uses "chown -R" on the RUNTIMEDIR in start_pre(), uses "chown" once in create_work_directory() if, and only if the working directory has been newly created.

@SoapGentoo @jlec : I think the updates are complete now.

### CODE REVIEW

**Comment by Yamakuzure** [comment]
@SoapGentoo @jlec : Sorry for the noise, but please do not merge just yet. I totally forgot about bug 599910 and want to change the revision bumps to updates. The desktop file has to be renamed from the default ${BIN}-${PN}.desktop to ${BIN}.desktop to fix the "Ugly Icon" issue for gnome users.

**Comment by Yamakuzure** [comment]
@SoapGentoo @jlec : Bug 599910 hotfixed, hotfix is tested and works. Hopefully on Gnome, too.

**Comment by Yamakuzure** [comment]
@SoapGentoo @jlec : Is there anything left open I have missed?

**Comment by Yamakuzure** [comment]
Michael Orlitzky just told me about "checkpath" from openrc-run, which can handle the RUNTIMEDIR with just one call.
I want to get rid of the ancient code in the init script that handles it and replace it with 'checkpath' before this PR is merged.
Thank you for your patience.

**Comment by Yamakuzure** [comment]
@SoapGentoo @jlec : Security fix finished and tested. I think we are good now, unless either of you finds anything else?

File: sci-misc/boinc/boinc-7.2.44-r2.ebuild [hunk 1]
```diff
@@ -118,6 +118,11 @@ src_install() {
 			newicon -s $s "${WORKDIR}"/boinc_${s}.png boinc.png
 		done
 		make_desktop_entry boincmgr "${PN}" "${PN}" "Math;Science" "Path=/var/lib/${PN}"
+
+		# Rename the desktop file to boincmgr.desktop to (hot)fix bug 599910
+		mv "${ED}"/usr/share/applications/boincmgr-${PN}.desktop \
+		   "${ED}"/usr/share/applications/boincmgr.desktop || \
+		   die "Failed to rename desktop file"
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.2.44-r2.ebuild]: shorten this to
`mv "${ED%/}"/usr/share/applications/boincmgr{-${PN},}.desktop || die "Failed to rename desktop file"`
always use ${ED%/} due to double-slashing (last section of https://devmanual.gentoo.org/ebuild-writing/variables/)

File: sci-misc/boinc/boinc-7.4.52-r2.ebuild [hunk 2]
```diff
@@ -123,6 +123,11 @@ src_install() {
 			newicon -s $s "${WORKDIR}"/boinc_${s}.png boinc.png
 		done
 		make_desktop_entry boincmgr "${PN}" "${PN}" "Math;Science" "Path=/var/lib/${PN}"
+
+		# Rename the desktop file to boincmgr.desktop to (hot)fix bug 599910
+		mv "${ED}"/usr/share/applications/boincmgr-${PN}.desktop \
+		   "${ED}"/usr/share/applications/boincmgr.desktop || \
+		   die "Failed to rename desktop file"
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.4.52-r2.ebuild]: same here

File: sci-misc/boinc/files/boinc.init [hunk 3]
```diff
@@ -54,6 +61,13 @@ env_check() {
 		return 1
 	fi
 
+	# The boinccmd is crucial, or we can not attach, suspend or resume
+	# the boinc client
+	if [ ! -x "$BOINCCMD" ] ; then
```
Comment (SoapGentoo) [inline sci-misc/boinc/files/boinc.init]: seeing that you're already using bashisms, prefer `[[ ]]` over `[ ]` as double brackets are more robust in bash than crappy single POSIX brackets

**Comment by SoapGentoo** [comment]
@Yamakuzure ping? for the last round of edits

**Comment by Yamakuzure** [comment]
@SoapGentoo : I am very sorry, but I missed that mail. It was there, though.

**Comment by Yamakuzure** [comment]
@SoapGentoo @jlec : I have implemented the latest change requirements. Hopefully I've got it right now. Sorry again for the delay!
